cpp/arcticdb/storage/mongo/mongo_client.cpp

```diff
@@ -29,7 +30,7 @@ namespace detail {
 template<typename ElementType>
 std::string get_string_element(const ElementType& element) {
 #if (MONGOCXX_VERSION_MAJOR * 1000 + MONGOCXX_VERSION_MINOR) >= 3007
-    return std::string(element.get_string());
+    return bsoncxx::string::to_string(element.get_string().value);
```

Comment:
I still wonder why this is necessary. BSONCXX's `b_string` has an [`operator stdx::string_view() const`](https://github.com/mongodb/mongo-cxx-driver/blame/master/src/bsoncxx/types.hpp#LL141C20-L141C54).

In the environments where `stdx == std`, [`std::string` ctor (10)](https://en.cppreference.com/w/cpp/string/basic_string/basic_string) should apply. If `stdx != std`, then `bsoncxx::string::to_string` will not return `std::string` so an implicit conversion is being done on return.